### PR TITLE
[MRG+1] DOC: sphinx-gallery histograms

### PR DESCRIPTION
**Pull request by choldgraf**
This attempts to consolidate a bunch of very short examples for histograms that were in the pylab folder into a longer-form example that's suitable for deploying with sphinx-gallery.

### CODE REVIEW

**Comment by choldgraf** [comment]
Note that there is some degree of redundancy between histogram examples in the `statistics` folder now. Lemme know if it's worth trying to consolidate further and I can put in some time to do so.

File: examples/statistics/plot_hist.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib import colors
+from matplotlib.ticker import PercentFormatter
+
+# Fixing random state for reproducibility
+np.random.seed(19680801)
+
+
+###############################################################################
```
Comment (NelleV) [inline examples/statistics/plot_hist.py]: For reviews, this syntax in sphinx gallery is supported to convert this into text in both the html and to a text block in the jupyter notebook version of the plot.

**Comment by NelleV** [comment]
My opinion is that we should take baby steps: this is already a great improvement to our examples and gallery. We can consolidate with other histogram examples later, though it is definitely on my TODO list.

**Comment by phobson** [comment]
This looks great. I'll come back to it can give it a more complete review later.

File: examples/statistics/plot_hist.py (hunk 1 above)
Comment (QuLogic) [inline examples/statistics/plot_hist.py]: Doesn't this cause a PEP8 error (No space after `#`)?

File: examples/statistics/plot_hist.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+# Generate data and plot a simple histogram
+# -----------------------------------------
+#
+# To generate a 1D histogram we only need a single vector of numbers. For a 2D
+# histogram we'll need a second vector. We'll generate both below, and show
+# the histogram for each vector
```
Comment (QuLogic) [inline examples/statistics/plot_hist.py]: Trailing period.

File: examples/statistics/plot_hist.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+
+###############################################################################
+# Updating histogram colors
+# -------------------------
+#
+# The histogram method returns (among other things) a `patches` object. This
+# gives us access to the properties of the objects drawn. Using this, we can
+# edit the histogram to our liking. Let's change the color of each bar
+# based on its y value
```
Comment (QuLogic) [inline examples/statistics/plot_hist.py]: Trailing period.

File: examples/statistics/plot_hist.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+fig, ax = plt.subplots(tight_layout=True)
+hist = ax.hist2d(x, y)
+
+###############################################################################
+# Customizing your histogram
+# --------------------------
+#
+# Customizing a 2D histogram is similar to the 1D case, you can control
+# visual components such as the bin size or color normalization
```
Comment (QuLogic) [inline examples/statistics/plot_hist.py]: Trailing period.

File: examples/statistics/plot_hist.py (hunk 1 above)
Comment (NelleV) [inline examples/statistics/plot_hist.py]: flake8 doesn't complain about it.

Comment (QuLogic) [inline examples/statistics/plot_hist.py]: OK, turns out these are E265 and E266, but we've got them in our global ignore list.

**Pull request merged**